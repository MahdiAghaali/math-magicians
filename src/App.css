.App,
#root {
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  align-content: center;
  flex-direction: column;
  font-size: 1.2rem;
}

.calculator {
  background-color: #cddce2;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  gap: 2px;
  padding: 5px;
  width: 250px;
}

.display {
  background-color: gray;
  color: white;
  grid-column: 1/ span 4;
  height: 40px;
  line-height: 40px;
  padding: 5px;
  text-align: right;
}

.button {
  text-align: center;
  height: 40px;
  line-height: 40px;
  transition: all 0.2s;
}

.button:hover {
  cursor: pointer;
}

.digit,
.other {
  background-color: #ececec;
}

.digit:hover,
.other:hover {
  background-color: white;
}

.operator {
  background-color: orange;
}

.operator:hover {
  background-color: rgb(255, 186, 58);
}

.zero {
  grid-column: 1/ span 2;
}
