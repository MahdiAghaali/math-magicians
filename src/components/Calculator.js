import React from 'react';
import Button from './Button';
import Display from './Display';

export default class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="calculator">
        <span className="display">
          {total}
          {operation}
          {next}
        </span>
        <button type="button" className="button other" onClick={this.test}>AC</button>
        <button type="button" className="button other" onClick={this.test}>+/-</button>
        <button type="button" className="button other" onClick={this.test}>%</button>
        <button type="button" className="button operator" onClick={this.test}>+</button>
        <button type="button" className="button digit" onClick={this.test}>1</button>
        <button type="button" className="button digit" onClick={this.test}>2</button>
        <button type="button" className="button digit" onClick={this.test}>3</button>
        <button type="button" className="button operator" onClick={this.test}>-</button>
        <button type="button" className="button digit" onClick={this.test}>4</button>
        <button type="button" className="button digit" onClick={this.test}>5</button>
        <button type="button" className="button digit" onClick={this.test}>6</button>
        <button type="button" className="button operator" onClick={this.test}>x</button>
        <button type="button" className="button digit" onClick={this.test}>7</button>
        <button type="button" className="button digit" onClick={this.test}>8</button>
        <button type="button" className="button digit" onClick={this.test}>9</button>
        <button type="button" className="button operator" onClick={this.test}>/</button>
        <button type="button" className="button digit zero" onClick={this.test}>0</button>
        <button type="button" className="button other" onClick={this.test}>.</button>
        <button type="button" className="button operator" onClick={this.test}>=</button>
      </div>
    );
  }
}
