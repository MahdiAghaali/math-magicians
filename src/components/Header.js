import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Math Magician</h1>
    <nav>
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/calculator">calculator</Link></li>
        <li><Link to="/quotes">quotes</Link></li>
      </ul>
    </nav>
  </header>
);

export default Header;
