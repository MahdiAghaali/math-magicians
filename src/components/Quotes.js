/* eslint-disable max-len */
import React from 'react';

const Quotes = () => (

  <section>
    <h2>
      Qutoe:
    </h2>
    <p>
      <q>don&apos;t do drugs kids</q>
      <i>--Mohi</i>
    </p>
  </section>
);

export default Quotes;
