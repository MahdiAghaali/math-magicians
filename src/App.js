import React from 'react';
import './App.css';
import {
  Routes,
  Route,
} from 'react-router-dom';
import Header from './components/Header';
import Home from './components/Home';
import Quotes from './components/Quotes';
import CalcSection from './components/CalcSection';

export default class App extends React.PureComponent {
  render() {
    return (
      <div className="App">
        <Header />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<CalcSection />} />
          <Route path="/quotes" element={<Quotes />} />
        </Routes>
      </div>
    );
  }
}
