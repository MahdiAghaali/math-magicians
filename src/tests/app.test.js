import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import App from '../App';
import Calculator from '../components/Calculator';
import Header from '../components/Header';
import Home from '../components/Home';
import Quotes from '../components/Quotes';

describe('operate(numberOne, numberTwo, operation)', () => {
  it('renders App correctly', () => {
    const tree = TestRenderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Calculator correctly', () => {
    const tree = TestRenderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Header correctly', () => {
    const tree = TestRenderer
      .create(<Router><Header /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Home correctly', () => {
    const tree = TestRenderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Quotes correctly', () => {
    const tree = TestRenderer
      .create(<Quotes />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
