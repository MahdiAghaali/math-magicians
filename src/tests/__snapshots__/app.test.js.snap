// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operate(numberOne, numberTwo, operation) renders App correctly 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Math Magician
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            calculator
          </a>
        </li>
        <li>
          <a
            href="/quotes"
            onClick={[Function]}
          >
            quotes
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <section>
    <h1>
      Welcome to Math Magician!
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </section>
</div>
`;

exports[`operate(numberOne, numberTwo, operation) renders Calculator correctly 1`] = `
<div
  className="calculator"
>
  <span
    className="display"
  />
  <button
    className="button other AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="button other"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="button other"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="button digit"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="button digit zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="button other"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`operate(numberOne, numberTwo, operation) renders Header correctly 1`] = `
<header>
  <h1>
    Math Magician
  </h1>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          calculator
        </a>
      </li>
      <li>
        <a
          href="/quotes"
          onClick={[Function]}
        >
          quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`operate(numberOne, numberTwo, operation) renders Home correctly 1`] = `
<section>
  <h1>
    Welcome to Math Magician!
  </h1>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
</section>
`;

exports[`operate(numberOne, numberTwo, operation) renders Quotes correctly 1`] = `
<section>
  <h2>
    Qutoe:
  </h2>
  <p>
    <q>
      don't do drugs kids
    </q>
    <i>
      --Mohi
    </i>
  </p>
</section>
`;
